### PR TITLE
Allow duplicate Persons

### LINKED ISSUE #45: Allow duplicate Persons

### PR DESCRIPTION
**Pull request by george-yeo**
- Added new PersonList that allows duplicates
- Modified AddressBook to use PersonList instead of UniquePersonList
- Updated test cases to reflect duplicates

Closes #45

### CODE REVIEW

**Review by ty4g1 — APPROVED** [review]
LGTM. 

If you have the time, maybe you could refactor the code so that `UniquePersonList` extends from `PersonList`, and only overrides the changed methods. Since there seems to be quite a bit of duplicated logic right now.

**Review by sooyitao — COMMENTED** [review]
Duplicate persons do not load  from storage after restarting app.

File: docs/UserGuide.md [hunk 1]
```diff
@@ -74,7 +74,7 @@ Format: `help`
 
 ### Adding a person: `add`
 
-Adds a person to the address book.
+Adds a person to the address book. People with same names are allowed.
```
Comment (sooyitao) [inline docs/UserGuide.md:77]: People with same number is also allowed.

File: src/main/java/seedu/address/model/person/PersonList.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    public void add(Person toAdd) {
+        requireNonNull(toAdd);
+
+        internalList.add(toAdd);
+    }
+
+    /**
+     * Replaces the person {@code target} in the list with {@code editedPerson}.
+     * {@code target} must exist in the list.
+     * The person identity of {@code editedPerson} must not be the same as another existing person in the list.
```
Comment (sooyitao) [inline src/main/java/seedu/address/model/person/PersonList.java:44]: Person can be the same as existing person.

**Pull request merged**